packages/dd-trace/src/platform/node/request.js

```diff
@@ -14,14 +14,16 @@ function request (options, callback) {
   }, options)
 
   const data = [].concat(options.data)
+  const client = options.protocol === 'https:' ? https : http
+  const agent = new client.Agent({ keepAlive: true })
```

Comment:
Fwiw when I've seen this in the past, the agent is declared outside of the request so that the instance/connection pool is reused across `request` invocations. Is that not necessary? I'm not 100% sure.

Reply:
In my tests that still worked, but you're right it could cause unexpected behavior so I'll move it out. Good catch 👍